api/desecapi/pdns_change_tracker.py

```diff
@@ -203,13 +214,10 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
         self.transaction.__exit__(None, None, None)
 
-        for domain_name, domain_pdns_id in {
-            (change.domain_name, change.domain_pdns_id)
-            for change in changes if change.axfr_required()
-        }:
-            _pdns_put(NSMASTER, '/zones/%s/axfr-retrieve' % domain_pdns_id)
-            domain = Domain.objects.get(name=domain_name)
-            domain.published = timezone.now()  # TODO consider moving this to the AXFR logic
+        for name in axfr_required:
+            _pdns_put(NSMASTER, '/zones/%s/axfr-retrieve' % pdns_id(name))
+            domain = Domain.objects.get(name=name)
+            domain.published = timezone.now()
```

Comment:
`Domain.objects.filter(name=name).update({'published': timezone.now())` (or similar), saves the `SELECT` query

Reply:
nice